### DISCUSSION TITLE
Fork tui-realm?

### DISCUSSION THREAD

**Discussion by hasezoey** [tramhao/termusic#361] [Ideas]
Seeing as [`tui-realm`](https://github.com/veeso/tui-realm) has been basically inactive for quite a while (no commits, ignoring the readme update; no comments to actual issues, comments to PRs) and having a new "major" version out, but us being blocked from upgrading to it due to sever issues (cursor event spam and low event polling rate), i have considered of forking tui-realm for us with those fixes applied and maybe some extra QOL changes.

What do you think @tramhao ?

PS: the same question for [`viuer`](https://github.com/atanunq/viuer) as it is blocking us from updating [`image`](https://github.com/tramhao/termusic/blob/1a38ac52111b83d2a86537283ee0bf804e2a3095/Cargo.toml#L45-L46)

**Comment by tramhao** [comment]
Considering  tui realm, I saw a new lib which is the successor of tui.rs. Maybe the effect is better? But we need basically rewrite termusic tui and it is a lot of work. For viuer, forking should be fine because it is a smaller package and easier to maintain. Anyway, I am open to both forks  and it happens to me before for mpv.rs , cpal and more.

**Reply by hasezoey** [reply]
> Considering tui realm, I saw a new lib which is the successor of tui.rs. Maybe the effect is better? But we need basically rewrite termusic tui and it is a lot of work

yes `ratatui` is the successor to `tui.rs`, which tuirealm added support for in `1.9.0`, which is the version we couldnt upgrade to.
aside from that, i dont see any big problems to changing from tuirs to ratatui (tuirealm `1.9.0` still has support for tuirs so we dont need to switch immediately).

**Comment by tramhao** [comment]
ratatui is much more popular and they have a control to show image. It could be an improvement compared with current implementation to show album photo.

**Comment by hasezoey** [comment]
As a update, since #375 we could upgrade to tui-realm 2.0.0, which fixes all the upgrade issues we had (and also migrated to to use the ratatui backend in tui-realm), also with #376 viuer gets upgraded so that image 0.25 could be used.

---

because it was brought up that we switch to ratatui directly, please note that ratatui does not have a component system and we would need to do that ourself, thought that would be a discussion for another thread.
